compiler/src/iree/compiler/Dialect/Flow/Transforms/DispatchWithTransformDialect.cpp

```diff
@@ -26,6 +30,33 @@ namespace iree_compiler {
 namespace IREE {
 namespace Flow {
 
+namespace {
+
+/// Builds transform IR forming dispatch regions for reductions.
+void buildReductionDispatch(ImplicitLocOpBuilder &builder, Value scopeH) {
+  auto pdlOperation = pdl::OperationType::get(builder.getContext());
+  SmallVector<Type> matchedTypes(4, pdlOperation);
+  auto matched = builder.create<transform_ext::MatchCallbackOp>(
+      matchedTypes, "reduction_partial",
+      transform::FailurePropagationMode::Suppress, scopeH);
+
+  auto takeFirst = builder.create<transform_ext::TakeFirstOp>(
+      pdlOperation, pdlOperation,
+      ValueRange{matched->getResults().back(),
+                 matched->getResults().drop_back().back()});
+  Value region = builder.create<transform_dialect::WrapInDispatchRegionOp>(
+      pdlOperation, takeFirst.getFirst());
+  region =
```

Comment:
All this could be a 1-liner if MovePrecedingOpIntoDispatchRegionOp was extended to allow unpacking the `getRest` handle on the fly (did that for some other op).

If you don't want to do it here, could you add to the TODO section of the running doc ?

Reply:
Not exactly one-liner, but changed this to merge handles + single move.

Reply:
ah .. I saw rest and thought tail .. 

Would be great to have some nicer API to hide all this mess but fine as a followup.